internal_macros/src/lib.rs

```diff
@@ -3,74 +3,194 @@
 //! Here be dragons 🐉
 
 extern crate proc_macro;
+extern crate proc_macro2;
 extern crate quote;
 extern crate syn;
 
+use std::iter::IntoIterator;
+
 use proc_macro::TokenStream;
+use proc_macro2::Span;
 use quote::quote;
-use syn::{Arm, Expr, ExprLit, Item, Lit, punctuated::Pair, Pat, PatLit, parse_macro_input};
-
-// We could include_str! inkwell's Cargo.toml and extract these features
-// so that they don't need to also be managed here...
-const ALL_FEATURE_VERSIONS: [&str; 8] = [
-    "llvm3-6",
-    "llvm3-7",
-    "llvm3-8",
-    "llvm3-9",
-    "llvm4-0",
-    "llvm5-0",
-    "llvm6-0",
-    "llvm7-0",
-];
-
-fn panic_with_usage() -> ! {
-    panic!("llvm_versions must take the form: X.Y => (X.Y || latest)");
-}
+use syn::parse_macro_input;
+use syn::parse::{Parse, ParseStream, Result, Error};
+use syn::{Token, LitFloat, Ident, Item};
+
+const FEATURES_ENV: &'static str = env!("INKWELL_FEATURES");
```

Comment:
Hmm since this all runs at compile time I'm a little confused about performance implications of const vs static here.

Normally const means it's evaluated once at compile time and inserted at all call sites... but we're already in compile time... so what happens? 😄

I wonder if a global variable is better here...
Maybe ideally `get_feature_versions` would be a const function, but those aren't stable yet.
Maybe I'm thinking about it too much. Just don't want a compile time bloat

Reply:
I would be very surprised if there was overhead using a const here, especially since it has one usage in `get_feature_versions`, so at worst it would get folded in to that function. In general though, it may be possible for it to be less ideal than a global static, but its going to depend heavily on what compiler settings are used (i.e. is it optimizing for speed or size) and how large the values are, since I suspect LLVM doesn't like to replace usages of constants with the constant value itself when that value is larger than a few words. The only way to know for sure though is to do some side-by-side comparisons.

As for why I chose const vs static here, it was just a semantic choice - the value is constant from the point of view of the module definition (i.e. when it is built, the value is fed in from the environment and it never changes). A static would also work just fine, and I think for our purposes there is likely no meaningful difference between the two, but until I know for sure, I will defer to your preference, just let me know how you'd like it to be set up :)